packages/host/app/components/matrix/payment-setup.gts

```diff
@@ -124,10 +135,10 @@ export default class PaymentSetup extends Component<Signature> {
             <BoxelButton
               @as='anchor'
               @kind='primary'
-              @href={{this.stripePaymentLink}}
+              @disabled={{this.setupPaymentMethod.isRunning}}
               data-test-setup-payment
               class='setup-button'
-              target={{if (eq environment 'development') '_blank' '_self'}}
+              {{on 'click' (perform this.setupPaymentMethod)}}
```

Comment:
Why does this have to be a task? Why not simply `href={{this.stripeFreePlanPaymentLink}}`?